src/Activity.php

```diff
@@ -38,13 +39,24 @@ public function getLeader(): Leader
     public function subscribe(Visitor $visitor, SubscriptionInterface $subscription): array
     {
         $activityVisits = [];
-        foreach ($this->schedule->materialize($subscription->getStartDay(), $subscription->getPeriod()) as $day) {
-            $activityVisits[] = new Visit($day, $this, $visitor);
+        try {
+            $subscription->isValid($this);
+            foreach ($this->schedule->materialize($subscription->getStartDay(), $subscription->getPeriod()) as $day) {
+                $activityVisits[] = new Visit($day, $this, $visitor);
+            }
+        } catch (SubscriptionIsForbidden $exception) {
```

Comment:
Don't catch the exception here, let it be a part of business logic: we'll catch it in some client code.

Reply:
Do I need to throw an exception forward? Otherwise, the subscription will be implemented

Reply:
Or completely exclude the try-catch from this code block?

Reply:
Yes, don't catch here anything.